pandas/tests/indexes/datetimes/test_insert.py

```diff
@@ -24,6 +24,19 @@ def test_insert_invalid_na(self, tz):
         with pytest.raises(TypeError, match="incompatible label"):
             idx.insert(0, np.timedelta64("NaT"))
 
+    def test_insert_empty_preserves_freq(self, tz_naive_fixture):
+        tz = tz_naive_fixture
+        dti = DatetimeIndex([], tz=tz, freq="D")
+        item = Timestamp("2017-04-05").tz_localize(tz)
+
+        result = dti.insert(0, item)
+        assert result.freq == dti.freq
```

Comment:
IIRC you mentioned that `assert_index_equal` doesn't compare freq i.e. we can't just simply use `assert_index_equal` here?

Reply:
correct.  ive got a branch that adds a freq check, am whittling down the actual bugs it identifies before making a PR for it.